test/authentication.js

```diff
@@ -1,131 +1,33 @@
-const should = require('should');
+/* global describe, it, before */
+
+require('should');
 
 const zapier = require('zapier-platform-core');
 
 const App = require('../index');
-const appTester = zapier.createAppTester(App);
-
-describe('basic auth app', () => {
-
-  it('auth succcessfully', (done) => {
-    zapier.tools.env.inject();
-    const bundle = {
-      authData: {
-        baseUrl: process.env.TEST_BASE_URL,
-        username: process.env.TEST_BASIC_AUTH_USERNAME,
-        password: process.env.TEST_BASIC_AUTH_PASSWORD
-      }
-    };
-
-    appTester(App.authentication.test, bundle)
-      .then((response) => {
-        response.status.should.eql(200);
-        should.exist(response.json.total);
-        should.not.exist(response.json.error);
-        done();
-      })
-      .catch(done);
-  }).timeout(15000);
-
-  it('fails auth on bad password', (done) => {
-    zapier.tools.env.inject();
-    const bundle = {
-      authData: {
-        baseUrl: process.env.TEST_BASE_URL,
-        username: process.env.TEST_BASIC_AUTH_USERNAME,
-        password: 'badpwd'
-      }
-    };
-
-    appTester(App.authentication.test, bundle)
-      .then(() => {
-        done('Should not get here');
-      })
-      .catch((error) => {
-        error.message.should.containEql('Authorization denied, invalid credentials.');
-        done();
-      });
-  }).timeout(15000);
-
-  it('fails auth on bad username', (done) => {
-    zapier.tools.env.inject();
-    const bundle = {
-      authData: {
-        baseUrl: process.env.TEST_BASE_URL,
-        username: 'badpwd',
-        password: process.env.TEST_BASIC_AUTH_PASSWORD
-      }
-    };
-
-    appTester(App.authentication.test, bundle)
-      .then(() => {
-        done('Should not get here');
-      })
-      .catch((error) => {
-        error.message.should.containEql('Authorization denied, invalid credentials.');
-        done();
-      });
-  }).timeout(15000);
-
-  it('fails auth on invalid url', (done) => {
```

Comment:
Can we keep this test?

Reply:
The reason is that during the very long and detail review of this integration before Zapier made it public they insisted on useful error messages for users who do something wrong. So we should test for these cases.

Reply:
@kuzmany did you see this comment?

Reply:
I don't know what do you expect from that tests. If oauth2 going to invalid url, then request failed always.
I've added fails auth on invalid url test, but let me know what do you want to add exactly.